qcore/shared.py

```diff
@@ -57,16 +57,17 @@ def get_corners(model_params, gmt_format=False):
     return corners, cnr_str
 
 
-def non_blocking_exe(
-    cmd, debug=True, shell=False, stdout=True, stderr=True, **kwargs
-):
-    # always split for consistency
-    if type(cmd) == str:
+def non_blocking_exe(cmd, debug=True, shell=False, stdout=True, stderr=True, **kwargs):
+    # always split for consistency (But only when shell == False
```

Comment:
This comment needs revision.

Reply:
> Can we get an example of where this was an issue?

I may be barking at the wrong tree, but vaguely remember when passing an argument surrounded by "", we used to have a problem.

Reply:
if the command is a list, instead of a string, the underlying behavior is different.
and this will cause issue if the shell/executable have a strict parsing-logic, in this case qsub with -v.
the values comes after -v and itself have to be parsed as one, and the easiest way to do so is execute the the command in one line with shell == True.

Reply:
So I was on the right track to guess that this issue is related to "". 

> qsub -v arg1="$var1/path",arg2='$foo',arg3=3 -otherflags script.sh

from https://wiki.canterbury.ac.nz/display/QuakeCore/PBS

Reply:
Ah yep. I think that argument should be handled differently to shell? As shell=True enables different functionality to the split/join argument handling.

To be honest splitting commands with .split() isn't the best practice in general - if we want to fix this properly it might be better to change the exe function to expect a list rather than a string?

Reply:
For this case I agree that the command should be submitted as a pre-split list, so that we can control which parts do and don't get split. Rather than changing how we process commands being submitted for execution

Reply:
the exe() can already handle cmd as a list.

Reply:
That's my point. Split the command earlier, not in here. IT means you can control how the split happens instead of just everything.
Making exe only accept lists is possibly a bit much, as it would result in a number of calls to split() further up the line.

Reply:
![image](https://user-images.githubusercontent.com/25143301/132619201-58fc55bc-c7e9-49e3-8078-871661e6b2d6.png)
I was thinking of instead of creating the commands instead of long strings to create them as lists.
e.g.
[wget, -O, zip_download_path, DATA_DOWNLOAD_PATH]

Reply:
double instead of...

Reply:
I also remembered an occasion where plot_items.py -t "This is a title"  (I think it was spawned by compare_runs.py screwed up. https://github.com/ucgmsim/visualization/blob/master/compare/compare_runs.py )

I ended up with plot_items.py -t This_is_a_title.  I think it will be nice if a solution here can address that too.